core/src/main/java/org/apache/james/core/builder/MimeMessageBuilder.java

```diff
@@ -398,7 +398,11 @@ public MimeMessage build() throws MessagingException {
 
         List<Header> headerList = headers.build();
         for (Header header: headerList) {
-            wrappedMessage.setHeader(header.name, header.value);
+            if (header.name.equals("Message-ID")) {
```

Comment:
the commit title "Introduce a Memoized supplier for Java8" is wrong

Reply:
Good reminder, thanks!